docs/user-interaction-config-file-spec.rst

```diff
@@ -0,0 +1,171 @@
+or ``KeyboardSpoke``.
+
+
+Section namespace
+-----------------
+
+Each section *must* contain the ``visited`` key with a value of either true if the user has visited
+the corresponding screen or false if not.
+
+Optional each section can contain one or more keys with the ``changed_`` prefix which track if the user
+has changed an option on the screen. If the option is changed by the user, the corresponding key is set
+to true. If the given option has not been changed by the user then the corresponding key can either be
+omitted or set to false.
+
+Example:
+
+::
+
+    [DatetimeSpoke]
+    visited=1
+    changed_timezone=1
+    changed_ntp=0
+    changed_timedate=1
+
+In this example the user has visited the date & time spoke and has changed the timezone & time/date,
+but not the NTP settings. Note that the ``changed_ntp`` key could also be omitted as the user has not changed
+the NTP options.
+
+Another example:
+
+::
+
+    [KeyboardSpoke]
+    visited=0
+
+Here the user has not visited the keyboard spoke and thus could not have changed any options,
+so all ``changed_*`` keys (if any) have been omitted.
+
+
+Full configuration file example
+-------------------------------
+
+::
+
+    post_install_tools_disabled=0
+
+    [DatetimeSpoke]
+    visited=1
+    changed_timezone=1
+    changed_ntp=0
+    changed_timedate=1
+
+    [KeyboardSpoke]
+    visited=0
+
+On the first line of the example the ``post_install_tools_disabled`` is equal to false, which means that
+post installation setup tools should proceed as usual. In this case the ``post_install_tools_disabled`` key
+might also be omitted.
+
+Next there are two sections corresponding to two screens - ``Datetimespoke`` and ``KeyboardSpoke``.
+
+The user has visited the date & time screen and has changed various options, but not the NTP settings.
+On the other hand the keyboard screen has not been visited at all.
+
+
+Parsing and writing the of the configuration file by Anaconda
+-------------------------------------------------------------
+
+If the user interaction file exists during Anaconda startup, it will be parsed and taken into account
+when deciding which screens to show during the installation. This make it possible for secondary
+installation setup tools to run before Anaconda and query the user for information.
+
+This can be for example a tool querying the user for language settings. Then once Anaconda starts it can
+skip the language selection screen as language has already been set by the tool.
+
+Once the installation process is done Anaconda will write out information about what screens the user has
+and has not visited and optionally which settings have been changed by the user.
+
+If Anaconda successfully parsed an existing user interaction configuration file, any valid settings present
+in the file will propagate to the configuration file when it is written-out by Anaconda.
+
```

Comment:
What about any comments in the file? We should probably assume no comments are preserved.